pallets/rewards/src/lib.rs

```diff
@@ -0,0 +1,546 @@
+			>,
+		) -> DispatchResult {
+			//check to ensure governance
+			T::GovernanceOrigin::ensure_origin(origin.clone())?;
+
+			//check if reward id present in storage
+			ensure!(
+				<InitializeRewards<T>>::contains_key(reward_id),
+				Error::<T>::RewardIdNotRegister
+			);
+
+			if let Some(reward_info) = <InitializeRewards<T>>::get(reward_id) {
+				//add all the beneficiary account in storage
+				for beneficiary in beneficiaries {
+					//calculate total rewards receive based on the conversion factor
+					let contribution: u128 = beneficiary.1.saturated_into();
+					let total_rewards_in_pdex: BalanceOf<T> = contribution
+						.saturating_mul(conversion_factor.saturated_into())
+						.saturating_div(UNIT_BALANCE)
+						.saturated_into();
+
+					if total_rewards_in_pdex > MIN_REWARDS_CLAIMABLE_AMOUNT.saturated_into() {
+						let reward_info = RewardInfoForAccount {
+							total_reward_amount: total_rewards_in_pdex,
+							claim_amount: 0_u128.saturated_into(),
+							is_initial_rewards_claimed: false,
+							is_initialized: false,
+							lock_id: REWARDS_LOCK_ID,
+							last_block_rewards_claim: reward_info.start_block,
+						};
+						<Distributor<T>>::insert(reward_id, beneficiary.0, reward_info);
+					} else {
+						//emit a event
+						Self::deposit_event(Event::UserRewardNotSatisfyingMinConstraint {
+							user: beneficiary.0,
+							amount_in_pdex: total_rewards_in_pdex,
+							reward_id,
+						});
+					}
+				}
+			} else {
+				//will not occur since we are already ensuring it above, still sanity check
+				return Err(Error::<T>::RewardIdNotRegister.into())
+			}
+			Ok(())
+		}
+
+		///The extrinsic will transfer and lock users rewards in the users account
+		/// Parameters,
+		/// origin: The users address which has been mapped to reward id
+		/// id: Reward id
+		#[pallet::weight(10_000)]
+		pub fn initialize_claim_rewards(origin: OriginFor<T>, reward_id: u32) -> DispatchResult {
+			let user: T::AccountId = ensure_signed(origin)?;
+			//check if given id valid or not
+			ensure!(
+				<InitializeRewards<T>>::contains_key(reward_id),
+				Error::<T>::RewardIdNotRegister
+			);
+
+			//check if user is added in reward list
+			ensure!(<Distributor<T>>::contains_key(reward_id, &user), Error::<T>::UserNotEligible);
+
+			<Distributor<T>>::mutate(reward_id, user.clone(), |user_reward_info| {
+				if let Some(user_reward_info) = user_reward_info {
+					// only unlock reward if current block greater than or equal to the starting
+					// block of reward
+					if let Some(reward_info) = <InitializeRewards<T>>::get(reward_id) {
+						ensure!(
+							reward_info.start_block.saturated_into::<u128>() <=
+								<frame_system::Pallet<T>>::block_number()
+									.saturated_into::<u128>(),
+							Error::<T>::RewardsCannotBeUnlockYet
+						);
+					}
+
+					//check if user already unlocked the rewards
+					ensure!(!user_reward_info.is_initialized, Error::<T>::RewardsAlreadyUnlocked);
```

Comment:
This assertion is not covered in tests

Reply:
Added it in `claim_rewards_at_start_block` test case